apollo-runtime/src/main/java/com/apollographql/apollo/ApolloClient.java

```diff
@@ -301,28 +311,38 @@ public Builder logger(@Nullable Logger logger) {
      * @return The configured {@link ApolloClient}
      */
     public ApolloClient build() {
-      checkNotNull(okHttpClient, "okHttpClient is null");
       checkNotNull(serverUrl, "serverUrl is null");
 
-      apolloLogger = new ApolloLogger(logger);
-      moshi = moshiBuilder.build();
+      ApolloLogger apolloLogger = new ApolloLogger(logger);
+      Moshi moshi = moshiBuilder.build();
 
-      if (httpCacheStore != null && httpEvictionStrategy != null) {
-        httpCache = new HttpCache(httpCacheStore, httpEvictionStrategy, apolloLogger);
-        okHttpClient = okHttpClient.newBuilder().addInterceptor(httpCache.interceptor()).build();
+      okhttp3.Call.Factory callFactory = this.callFactory;
+      boolean localClient = this.callFactory == null;
+      if (localClient) {
+        callFactory = new OkHttpClient();
+      }
+
+      if (httpCache != null && callFactory instanceof OkHttpClient) {
+        if (localClient) {
+          apolloLogger.w("Created local client, adding passed in HttpCache");
```

Comment:
That's confusing as we don't create actually local client, we just add interceptor to it.
Maybe it will be better to change it to smth like this:
```
apolloLogger.w("Add passed in HttpCache to provided http client");
```